ngraph/core/reference/src/runtime/reference/random_uniform.cpp

```diff
@@ -0,0 +1,222 @@
+}
+
+double uint32_to_double(uint32_t x1, uint32_t x2) {
+    uint64_t significant = ((static_cast<uint64_t>(x1) & 0xfffffu) << 32) | static_cast<uint64_t>(x2);
+    uint64_t x_uint64 = ((static_cast<uint64_t>(1023) << 52) | significant);
+
+    double x_double;
+    memcpy(&x_double, &x_uint64, sizeof(x_uint64));
+    return x_double - 1.0;
+}
+
+uint64_t uint32_to_uint64(uint32_t x1, uint32_t x2) {
+    return (static_cast<uint64_t>(x2) << 32) | static_cast<uint64_t>(x1);
+}
+
+bfloat16 uint32_to_bfloat16(uint32_t x) {
+    uint16_t x_uint16 = static_cast<uint16_t>(x);
+    x_uint16 = (static_cast<uint16_t>(127) << 7) | (x_uint16 & 0x7fu);
+
+    bfloat16 x_bfloat16;
+    memcpy(&x_bfloat16, &x_uint16, sizeof(x_uint16));
+    return x_bfloat16 - static_cast<bfloat16>(1);
+}
+
+void run_philox(uint64_t key, uint64_t counter, uint64_t n, size_t n_rounds, std::vector<uint32_t>& res) {
+    for (size_t i = 0; i < n_rounds; i++) {
+        calculate_round(key, counter, n);
+        if (i < n_rounds - 1)
+            raise_key(key);
+    }
+    auto res1 = split_high_low(n);
+    auto res2 = split_high_low(counter);
+    res[0] = res1.first;
+    res[1] = res1.second;
+    res[2] = res2.first;
+    res[3] = res2.second;
+}
+
+void random_uniform(const uint64_t* out_shape,
+                    const char* min_val,
+                    const char* max_val,
+                    char* out,
+                    const Shape& out_shape_shape,
+                    ngraph::element::Type elem_type,
+                    uint64_t seed,
+                    uint64_t seed2) {
+    if (seed == 0 && seed2 == 0) {
+        std::srand(std::time(nullptr));
+        seed = std::rand();
+    }
+    uint64_t key = seed;
+    uint64_t counter = seed2;
+    uint64_t n = 0;
+    size_t shape_count = shape_size(out_shape_shape);
+    size_t elem_count = 1;
+    for (size_t i = 0; i < shape_count; i++) {
+        elem_count *= out_shape[i];
+    }
+    size_t step = elem_type.size() > 4 ? 2 : 4;
+    for (size_t k = 0; k < elem_count; k += step) {
+        // generate 4 random uint32 values using Philox algorithm
+        std::vector<uint32_t> res(4);
+        run_philox(key, counter, n, 10, res);
+        switch (elem_type) {
+        case ngraph::element::Type_t::f32: {
+            float res_float[4];
+            std::transform(res.data(), res.data() + 4, res_float, uint32_to_float);
+            float mn[1];
+            float mx[1];
+            memcpy(mn, min_val, elem_type.size());
+            memcpy(mx, max_val, elem_type.size());
+            // convert uint32 values to float32 and normalize to range
+            // [min_val, max_val)
+            std::transform(res.data(), res.data() + 4, res_float, [&mn, &mx](uint32_t elem) {
+                return uint32_to_float(elem) * (mx[0] - mn[0]) + mn[0];
+            });
+
+            memcpy(out + k * elem_type.size(), res_float, std::min((size_t)4, elem_count - k) * elem_type.size());
```

Comment:
magic const

Reply:
Added const name.

Reply:
Here I can use 'step' value instead of const. Corrected.